dask/dataframe/io/parquet/arrow.py

```diff
@@ -1382,127 +1175,251 @@ def _update_metadata_options(
         if gather_statistics is None:
             gather_statistics = bool(stat_col_indices)
 
-        return (
-            gather_statistics,
-            split_row_groups,
-            stat_col_indices,
-        )
+        # Get/transate filters
+        ds_filters = None
+        if filters is not None:
+            ds_filters = pq._filters_to_expression(filters)
+
+        # Define subset of `dataset_info` required by _collect_file_parts
+        dataset_info_kwargs = {
+            "fs": fs,
+            "split_row_groups": split_row_groups,
+            "gather_statistics": gather_statistics,
+            "partitioning_method": partitioning_method,
+            "filters": filters,
+            "ds_filters": ds_filters,
+            "schema": schema,
+            "stat_col_indices": stat_col_indices,
+            "aggregation_depth": aggregation_depth,
+            "chunksize": chunksize,
+            "partitions": partitions,
+        }
+
+        # Main parts/stats-construction
+        if has_metadata_file or metadata_task_size == 0:
+            # We have a global _metadata file to work with.
+            # Therefore, we can just loop over fragments on the client.
+
+            # Start with sorted (by path) list of file-based fragments
+            file_frags = sorted(
+                [frag for frag in ds.get_fragments(ds_filters)],
+                key=lambda x: natural_sort_key(x.path),
+            )
+            parts, stats = cls._collect_file_parts(file_frags, dataset_info_kwargs)
+        else:
+            # We DON'T have a global _metadata file to work with.
+            # We should loop over files in parallel
+
+            # Collect list of file paths.
+            # If valid_paths is not None, the user passed in a list
+            # of files containing a _metadata file.  Since we used
+            # the _metadata file to generate our dataset object , we need
+            # to ignore any file fragments that are not in the list.
+            all_files = sorted(ds.files, key=natural_sort_key)
+            if valid_paths:
+                all_files = [
+                    filef
+                    for filef in all_files
+                    if filef.split(fs.sep)[-1] in valid_paths
```

Comment:
This is not new code, but I'm sure it *could* be simplified in a simple stand-alone PR. We are essentially comparing the file names to a list of "valid" filenames that have not been filtered out during dataset discovery. We only take the last element, because that is the file name.